### PR TITLE
Feature: Recommendation UI

### LINKED ISSUE #290: Build a results layout with distinct categorization sections (Top Fit, Caution, Not Suitable).
### Description
Design a clean UI to visually separate highly recommended species from those that are likely to fail.

### Definition of Done

- [ ] Code Style adheres to `CONTRIBUTING.md` guidelines
- [ ] Code Comments: Complex logic is commented
- [ ] Clean Build: Feature branch merges cleanly with `master` (no conflicts).
- [ ] Unit/automated tests created and passing
- [ ] Documentation updated (if applicable)
- [ ] Local Data locations documented (if applicable)
- [ ] Task in Planner is linked to the relevant Pull Request.
- [ ] Reviewed and approved

---

### LINKED ISSUE #343: US-049: Interactive recommendation
### User Story

**As a**  
Inspector or Farmer

**I want**  
to view detailed species recommendations.

**So that**  
I can easily interpret suitability via an interface.

---

### Description  
This story targets Inspector or Farmers to view detailed species recommendations. It ensures I can easily interpret suitability via a visual layout.

---

### Acceptance Criteria  

- [x] Results layout built with distinct categorisation sections, including Top Fit, Caution, and Not Suitable.  

---

### PR DESCRIPTION
**Pull request by dwstephens**
## Description
This PR adds a recommendation UI to the front end, replacing the placeholder page. This implementation focuses on function over form, so we have something to show the PO and gather feedback for further improvements/changes.

I have followed a Container/Presentational pattern combined with the Custom Hook pattern for the architecture, as this appears to be the pattern applied in US-001.

To make this implementation work, I have updated the AuthContext to provide a function to allow access to the stored token. This change enables all other UI's to access the token when required.

## Related Issue / Task
Closes #343 closes #290 

## Team / Area

- [x] Frontend
- [ ] Backend
- [ ] Data Science
- [ ] GIS
- [ ] Documentation
- [ ] Other (please specify):

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Refactoring / maintenance
- [ ] Other (please describe):

## Checklist

- [x] I have performed a self-review of my code
- [x] My code follows the project’s style guidelines (linting & formatting)
- [x] This PR is based on the latest `upstream/master`
- [x] I have updated documentation if necessary

## Additional Notes
I have not changed the process of logging in. For now, to test this page, you need to log in as an administrator. This will redirect you to the admin page. Manually change the URL back to the home page, then select the recommendation link from there. Since you're logged in as an admin, you can profile any farm_id. 

I have not added snapshot tests. Once we have feedback from the PO, any additional changes and further tests will be a future story.

### CODE REVIEW

File: frontend/src/components/recommendations/recommendationSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  isLoading: boolean;
+}
+
+export default function RecommendationSearch({
+  onSearch,
+  isLoading,
+}: RecommendationSearchProps) {
+  const [searchInput, setSearchInput] = useState("");
+
+  const handleSearch = () => {
```
Comment (Naresh-Barua) [inline frontend/src/components/recommendations/recommendationSearch.tsx:15]: Small validation mismatch here: `handleSearch()` checks `searchInput.trim()`, but the button disabled state uses `!searchInput`.

That means whitespace-only input can still enable the button even though the search will not run. It may be cleaner to use the same trimmed check in both places for consistent behaviour.

File: frontend/src/components/recommendations/recommendationRow.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                  backgroundColor: getBarColor(item.score_mcda),
+                }}
+              ></div>
+            </div>
+          </div>
+        </td>
+        <td className="rec-td">
+          <button className="rec-details-btn" onClick={onToggle}>
+            {isExpanded ? "Hide" : "Details"}
+          </button>
```
Comment (Naresh-Barua) [inline frontend/src/components/recommendations/recommendationRow.tsx]: Nice interaction pattern here. One small accessibility improvement would be to add `aria-expanded={isExpanded}` to the button.

That would make the expand/collapse state clearer for screen readers.

File: frontend/src/hooks/useRecommendations.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          headers: {
+            Authorization: `Bearer ${token}`,
+            Accept: "application/json",
+          },
+        });
+
+        if (!response.ok) {
+          const errorData = await response.text();
+          throw new Error(errorData);
+        }
```
Comment (Naresh-Barua) [inline frontend/src/hooks/useRecommendations.ts:62]: The fetch/error handling looks clean, but I had a question about the user-facing behaviour.

Since errors are forwarded through `useAsyncError` and `hasSearched` is only set on success, does the page show a visible error state on failure? It looks like it may remain in the pre-search state instead.

File: frontend/src/test/recommendationComponents.test.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            isExpanded={true}
+            onToggle={vi.fn()}
+          />
+        </tbody>
+      </table>
+    );
+
+    expect(screen.getByText(/KEY FACTORS/i)).toBeInTheDocument();
+  });
+});
```
Comment (Naresh-Barua) [inline frontend/src/test/recommendationComponents.test.tsx:198]: The component tests cover the main recommendation table flow well. It may also be worth adding a small test for `ExcludedTable` / `ExcludedRow`, especially for expand/collapse behaviour and the empty state, since that is also user-facing logic introduced in this PR.

File: frontend/src/hooks/useRecommendations.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  const { getAccessToken } = useAuth();
+  const token = getAccessToken();
+  const throwAsyncError = useAsyncError();
+
+  const [recs, setRecs] = useState<Recommendation[]>([]);
+  const [excludes, setExcludes] = useState<ExcludedSpecies[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasSearched, setHasSearched] = useState(false);
+
+  useEffect(() => {
```
Comment (Naresh-Barua) [inline frontend/src/hooks/useRecommendations.ts]: Quick question on the token flow: since `getAccessToken()` is called during render and then used inside the effect dependency list, would it be cleaner to read the token inside the effect itself?

Not a blocker, just wondering if that would make the dependency behaviour a bit more explicit.

File: frontend/src/components/recommendations/recommendationSearch.tsx (hunk 1 above)
Comment (dwstephens) [reply frontend/src/components/recommendations/recommendationSearch.tsx:15]: @Naresh-Barua fixed with commit b079e4b.

File: frontend/src/components/recommendations/recommendationRow.tsx (hunk 2 above)
Comment (dwstephens) [reply frontend/src/components/recommendations/recommendationRow.tsx]: @Naresh-Barua  fixed with 1b4bf0. Also applied the change to the ExcludedRow component.

File: frontend/src/hooks/useRecommendations.ts (hunk 3 above)
Comment (dwstephens) [reply frontend/src/hooks/useRecommendations.ts:62]: @Naresh-Barua I refactored the useRecommendations to return a local error state instead of crashing the app via the Global Error boundary. Errors are now rendered on the page. This also meant I could remove the useAsyncError hook. Resolved in Commit dd9979

File: frontend/src/test/recommendationComponents.test.tsx (hunk 4 above)
Comment (dwstephens) [reply frontend/src/test/recommendationComponents.test.tsx:198]: @Naresh-Barua, I have added tests for the ExcludeTable and ExcludeRow. Also added checking for the aria-expand attribute on the buttons. Resolved with commits ff71a1 and d7ce24

File: frontend/src/hooks/useRecommendations.ts (hunk 5 above)
Comment (dwstephens) [reply frontend/src/hooks/useRecommendations.ts]: @Naresh-Barua, thanks for asking this question. It made me think about the possible issues. I have moved, getting the token inside the effect. This was possible because the getAccessToken() returns a stable function reference. This now ensures the token is always fresh. Resolved with commit b242ed.

**Comment by dwstephens** [comment]
@Naresh-Barua, thanks for your feedback. Has really helped me dig a little deeper into React and improve my code. I believe I have addressed all of your comments.

**Review by Naresh-Barua — APPROVED** [review]
Thanks for addressing all the feedback — changes look good now. Nice improvements on error handling and test coverage as well.
